### ISSUE TITLE
Unable to wrap Material.Components.Maui components

### ISSUE THREAD

**Issue by AlexeyZavar** [adospace/reactorui-maui#31]
Hi there. Trying to generate wrapper for the [Material.Components.Maui](https://github.com/mdc-maui/mdc-maui) library, but the generated code is invalid.
![image](https://user-images.githubusercontent.com/17935037/221905395-74d8c742-8c5e-41ff-9a7c-2112fe1a4f76.png)
![image](https://user-images.githubusercontent.com/17935037/221905482-15bda2c6-30b8-41fd-9ae5-123d0cc6d14e.png)
![image](https://user-images.githubusercontent.com/17935037/221905570-24a7260d-245c-4c53-9830-6024a9e18735.png)

As I see ([1](https://github.com/mdc-maui/mdc-maui/blob/master/Material.Components.Maui/Components/Button/Button.cs), [2](https://github.com/mdc-maui/mdc-maui/blob/7369c8f2d720f09d66a01e4fd9006d7b9f4b1080/Material.Components.Maui/Components/SKCanvasView/SKTouchCanvasView.cs)), they inherit from `SKTouchCanvasView`, not from `ContentView` or else.
Is it possible to generate wrappers for this library? Maybe with some workarounds in generator?

**Comment by adospace** [comment]
you have to generate also the base class:
[Scaffold(typeof(SkTouchCanvasView))]

and any other class up to a standard MAUI class like View

**Comment by adospace** [comment]
this for example is the generation code for the Maps control:

```c#
[Scaffold(typeof(Microsoft.Maui.Controls.Maps.Map), implementItemTemplate: true)]
[ScaffoldChildren(typeof(Microsoft.Maui.Controls.Maps.Pin), nameof(Microsoft.Maui.Controls.Maps.Map.Pins))]
[ScaffoldChildren(typeof(Microsoft.Maui.Controls.Maps.Polygon), nameof(Microsoft.Maui.Controls.Maps.Map.MapElements))]
[ScaffoldChildren(typeof(Microsoft.Maui.Controls.Maps.Polyline), nameof(Microsoft.Maui.Controls.Maps.Map.MapElements))]
[ScaffoldChildren(typeof(Microsoft.Maui.Controls.Maps.Circle), nameof(Microsoft.Maui.Controls.Maps.Map.MapElements))]
public partial class Map
{
}

[Scaffold(typeof(Microsoft.Maui.Controls.Maps.Pin))]
public partial class Pin
{
}

[Scaffold(typeof(Microsoft.Maui.Controls.Maps.MapElement))]
public partial class MapElement
{
}

[Scaffold(typeof(Microsoft.Maui.Controls.Maps.Polygon))]
public partial class Polygon
{
}

[Scaffold(typeof(Microsoft.Maui.Controls.Maps.Polyline))]
public partial class Polyline
{
}

[Scaffold(typeof(Microsoft.Maui.Controls.Maps.Circle))]
public partial class Circle
{
}
```

**Comment by AlexeyZavar** [comment]
Got it compiling, thanks.

Just in case somebody need it:
```csharp
using MauiReactor;

namespace Project.MaterialGenerated;

[Scaffold(typeof(Material.Components.Maui.Button))]
public partial class Button
{

}

[Scaffold(typeof(Material.Components.Maui.SKTouchCanvasView))]
public partial class SKTouchCanvasView
{

}

[Scaffold(typeof(SkiaSharp.Views.Maui.Controls.SKCanvasView))]
public partial class SKCanvasView
{

}
```

It's worth saying that target class (`Button` in my case) should be called exactly as source (`Material.Components.Maui.Button`), or there will be 2 classes.

Now I'm struggling with `Java.Lang.OutOfMemoryError: Exception of type 'Java.Lang.OutOfMemoryError' was thrown.` [error](https://github.com/mdc-maui/mdc-maui/issues/7), though styles are added correctly (`app.Resources.MergedDictionaries.Add(new MaterialStyles());` in `UseMauiReactorApp`). Don't really know if it's related to Reactor, but seems to be working on "stock" Maui.

**Comment by adospace** [comment]
I'll create a sample project using that library and I'll let you know

**Comment by adospace** [comment]
I tried to create a standard maui project using their library following the official guide (including point 4): https://mdc-maui.github.io/getting-started but it throws the Java.Lang.OutOfMemoryError exception on starting up.
I think it's a mdc-maui library problem so I'm going to close this thicket, please open a new one if you have other info on this issue.

**Comment by adospace** [comment]
After further investigation, I found that MauiReactor had some bugs exploited by the mdc controls integration that I've now resolved in the new version. I've done some tests and it works fine now. 

Still not sure of some mdc design decisions, for example, they set Vertical and Horizontal LayoutOptions inside styles. Also, I think they need to make more robust the measure pass more in order to avoid the creation of a canvas of infinite size leading to the above OOM. Anyway, given that one loads the resource dictionary containing the styles and pays attention to the layout options everything should work just fine.

BTW Your report helped a lot, thanks
